http/src/main/java/io/micronaut/http/body/MessageBodyWriter.java

```diff
@@ -89,13 +90,54 @@ default boolean isBlocking() {
      * @return The charset
      */
     static @NonNull Charset getCharset(@NonNull Headers headers) {
+        return findCharset(headers).orElse(StandardCharsets.UTF_8);
+    }
+
+    /**
+     * Resolve the charset.
+     *
+     * @param mediaType The mediaType
+     * @param headers   The headers
+     * @return The charset
+     */
+    static @NonNull Charset getCharset(@Nullable MediaType mediaType, @NonNull Headers headers) {
```

Comment:
should these be moved to utility methods of `HttpHeaders`?

Reply:
Not sure, I think in v5 we should change both read/writers to use `HttpHeaders` instead of `Headers`